linkerd/tls/src/server/mod.rs

```diff
@@ -159,17 +160,38 @@ where
 
         match self.local_identity.as_ref() {
             Some(local) => {
-                let config = Param::<Config>::param(local);
-                let local_id = Param::<LocalId>::param(local);
-                let timeout = tokio::time::sleep(self.timeout);
+                let config: Config = local.param();
+                let LocalId(local_id) = local.param();
 
+                let timeout = tokio::time::sleep(self.timeout);
                 Box::pin(async move {
-                    let (peer, io) = tokio::select! {
-                        res = detect(io, config, local_id) => { res? }
+                    // Detect the SNI from a ClientHello (or timeout).
+                    let (sni, io) = tokio::select! {
+                        res = detect_sni(io) => { res? }
                         () = timeout => {
                             return Err(DetectTimeout(()).into());
                         }
                     };
```

Comment:
nit/tioli (and i realize this code was already there): i wonder if this is expressed more concisely with [`tokio::time::timeout`](https://docs.rs/tokio/1.4.0/tokio/time/fn.timeout.html)? but, maybe we would still have to unpack a nested `Result` so it might be the same number of lines...